### ISSUE TITLE
Request parameter duplicated in request/response snippets

### ISSUE THREAD

**Issue by moritzzimmer** [spring-projects/spring-restdocs#316]
I'm trying to document a request parameter using `RestDocumentationRequestBuilders.get` using the following code (version `1.1.2.RELEASE`):

```java
mockMvc.perform(get("/content/{id}?show-related=true", "123")
                .header("Authorization", "Basic dXNlcjpwYXNzd29yZA==")
                .accept(MediaType.APPLICATION_JSON))
                .andExpect(status().isOk())
                .andExpect(content().contentTypeCompatibleWith(MediaType.APPLICATION_JSON))
                .andExpect(content().string("{\"content\":{\"foo\":\"bar\"},\"related\":[]}"))
                .andExpect(header().string("Cache-Control", "max-age=10"))
                .andDo(
                        document("content/id",
                                requestHeaders(
                                        headerWithName("Authorization").description("Basic auth credentials"),
                                        headerWithName("Accept").description("(optional) acceptable media type").optional()
                                ),
                                responseHeaders(
                                        headerWithName("Cache-Control").description("Cache-Control max-age TTL in seconds")
                                ),
                                pathParameters(
                                        parameterWithName("id").description("Unique id of the content to retrieve")
                                ),
                                requestParameters(
                                        parameterWithName("show-related").description("(optional) include related content in response.")
                                )
                        )
                );
```

The request param is duplicated in the generated `curl` and `http-request` snippets. Example:
```bash
$ curl 'https://my-host.com/content/123?show-related=true?show-related=true' -i -u 'user:password' -H 'Accept: application/json'
```
Although the `httpie` snippet is correct:
```bash
$ http --auth 'user:password' GET 'https://my-host.com/content/123?show-related=true' 'Accept:application/json'
```
Something wrong with my MockMvc code?

**Comment by wilkinsona** [comment]
That sounds a lot like #286 but it should be fixed in 1.1.2.RELEASE. Are you sure that's the version you're using? Specifically, are you sure you're using 1.1.2.RELEASE of `spring-restdocs-core`? If you are, can you please provide a small sample that reproduces the problem and I'll take a look.

**Comment by moritzzimmer** [comment]
@wilkinsona I declared `org.springframework.restdocs:spring-restdocs-mockmvc:1.1.2.RELEASE` using the dependency management plugin of `org.springframework.boot:spring-boot-gradle-plugin:1.4.0.RELEASE`. Unfortunately this pull in `org.springframework.restdocs:spring-restdocs-core:1.1.1.RELEASE`:

```
+--- org.springframework.restdocs:spring-restdocs-mockmvc: -> 1.1.2.RELEASE
|    +--- org.springframework:spring-test:4.2.5.RELEASE -> 4.3.2.RELEASE (*)
|    +--- org.springframework:spring-webmvc:4.2.5.RELEASE -> 4.3.2.RELEASE (*)
|    \--- org.springframework.restdocs:spring-restdocs-core:1.1.2.RELEASE -> 1.1.1.RELEASE
|         +--- javax.servlet:javax.servlet-api:3.1.0
|         +--- org.springframework:spring-web:4.2.5.RELEASE -> 4.3.2.RELEASE (*)
|         \--- com.fasterxml.jackson.core:jackson-databind:2.4.6.1 -> 2.8.1 (*)
```

Explicitly declaring both fixed the problem as you suggested
```
dependencyManagement {
    dependencies {
        dependency 'org.springframework.restdocs:spring-restdocs-mockmvc:1.1.2.RELEASE'
        dependency 'org.springframework.restdocs:spring-restdocs-core:1.1.2.RELEASE'
    }
}
```

Thx for your support.
